packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart

```diff
@@ -361,52 +323,48 @@ class HttpResponseView extends StatelessWidget {
   }
 }
 
-class HttpTextResponseViewer extends StatefulWidget {
+class HttpTextResponseViewer extends StatelessWidget {
   const HttpTextResponseViewer({
     super.key,
     required this.contentType,
     required this.responseBody,
     required this.currentResponesNotifier,
     required this.textStyle,
   });
+
   final String? contentType;
   final String responseBody;
-  final ValueListenable<NetworkResponseType> currentResponesNotifier;
+  final ValueListenable<NetworkResponseViewType> currentResponesNotifier;
   final TextStyle textStyle;
 
-  @override
-  State<HttpTextResponseViewer> createState() => _HttpTextResponseViewerState();
-}
-
-class _HttpTextResponseViewerState extends State<HttpTextResponseViewer> {
-  late NetworkResponseType currentLocalResponseType;
-
-  void setCurrentLocalResponseType(NetworkResponseType type) {
-    if (type == NetworkResponseType.auto) {
-      if (widget.contentType != null &&
-          widget.contentType!.contains('json') &&
-          widget.responseBody.isNotEmpty) {
-        currentLocalResponseType = NetworkResponseType.json;
-      } else {
-        currentLocalResponseType = NetworkResponseType.text;
-      }
-      return;
-    }
-    currentLocalResponseType = type;
-  }
-
   @override
   Widget build(BuildContext context) {
     return ValueListenableBuilder(
-      valueListenable: widget.currentResponesNotifier,
+      valueListenable: currentResponesNotifier,
       builder: (_, currentResponseType, __) {
+        late final NetworkResponseViewType currentLocalResponseType;
+
+        void setCurrentLocalResponseType(NetworkResponseViewType type) {
+          if (type == NetworkResponseViewType.auto) {
+            if (contentType != null &&
+                contentType!.contains('json') &&
+                responseBody.isNotEmpty) {
+              currentLocalResponseType = NetworkResponseViewType.json;
+            } else {
+              currentLocalResponseType = NetworkResponseViewType.text;
+            }
+            return;
+          }
+          currentLocalResponseType = type;
+        }
+
         setCurrentLocalResponseType(currentResponseType);
```

Comment:
This can be simplified further and then we can get rid of the setCurrentLocalResponseType method:
```dart 
var responseType = currentResponseType;
if (currentResponseType =- NetworkResponseViewType.auto) {
   if (contentType != null &&
         contentType!.contains('json') &&
         responseBody.isNotEmpty) {
     currentLocalResponseType = NetworkResponseViewType.json;
   } else {
     currentLocalResponseType = NetworkResponseViewType.text;
   }
}
```